e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,529 @@
+			Address: string(newMM2.FormattedAddress()),
+		},
+	}
+
+	require.Equal(t, expectedGetMasterMinterResponse.MasterMinter, showMMRes.MasterMinter)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Owner)
+
+	// ACTION: Update Master Minter to blacklisted Master Minter account
+	// EXPECTED: Success; Master Minter updated
+	// Status:
+	// 	Master Minter: newMM2
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "new-mm-3", math.OneInt(), noble)
+	newMM3 := w[0]
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, newMM3)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-master-minter", newMM3.FormattedAddress())
+	require.NoError(t, err, "failed to broadcast update-master-minter message")
+
+	showMMRes, err = showMasterMinter(ctx, val)
+	require.NoError(t, err, "failed to query show-master-minter")
+	expectedGetMasterMinterResponse = fiattokenfactorytypes.QueryGetMasterMinterResponse{
+		MasterMinter: fiattokenfactorytypes.MasterMinter{
+			Address: string(newMM3.FormattedAddress()),
+		},
+	}
+	require.Equal(t, expectedGetMasterMinterResponse.MasterMinter, showMMRes.MasterMinter)
+}
+
+func TestFiatTFConfigureMinterController(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// ACTION: Configure Minter Controller while TF is paused
+	// EXPECTED: Success; Minter Controller is configured with Minter
+
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "minter-controller-1", math.OneInt(), noble)
+	minterController1 := w[0]
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "minter-1", math.OneInt(), noble)
+	minter1 := w[0]
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	_, err := val.ExecTx(ctx, nw.fiatTfRoles.MasterMinter.KeyName(), "fiat-tokenfactory", "configure-minter-controller", minterController1.FormattedAddress(), minter1.FormattedAddress())
+	require.NoError(t, err, "error configuring minter controller")
+
+	showMCRes, err := showMinterController(ctx, val, minterController1)
+	require.NoError(t, err, "failed to query show-minter-controller")
+	expectedShowMinterController := fiattokenfactorytypes.QueryGetMinterControllerResponse{
+		MinterController: fiattokenfactorytypes.MinterController{
+			Minter:     minter1.FormattedAddress(),
+			Controller: minterController1.FormattedAddress(),
+		},
+	}
+	require.Equal(t, expectedShowMinterController.MinterController, showMCRes.MinterController)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Configure Minter Controller from non Master Minter account
+	// EXPECTED: Request fails; Minter Controller not configured with Minter
+	// Status:
+	// 	minterController1 -> minter1
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "default", math.OneInt(), noble, noble, noble)
+	minterController2 := w[0]
+	minter2 := w[1]
+	alice := w[2]
```

Comment:
just curious, why can you use one `GetAndFundTestUsers` to get multiple addresses here, whereas in the case above (lines 126-129) it's like one call produces one address?

Reply:
I should have written it like the lines highlighted here. It is a variadic function. Updated!